src/mapml.js

```diff
@@ -3132,53 +3450,57 @@ M.templatedTileLayer = function(template, options) {
 M.MapMLStaticTileLayer = L.GridLayer.extend({
 
   initialize: function (options) {
-    let zoomBounds = this._getZoomBounds(options.tileContainer,options.maxZoomBound);
-    options.maxNativeZoom = zoomBounds.nMax;
-    options.minNativeZoom = zoomBounds.nMin;
-    options.maxZoom = zoomBounds.max;
-    options.minZoom = zoomBounds.min;
+    this.zoomBounds = this._getZoomBounds(options.tileContainer,options.maxZoomBound);
+    L.extend(options, this.zoomBounds);
     L.setOptions(this, options);
     this._groups = this._groupTiles(this.options.tileContainer.getElementsByTagName('tile'));
   },
 
   onAdd: function(){
-    this._bounds = this._getLayerBounds(this._groups,this._map.options.crs.options.resolutions); //stores meter values of bounds
-    this.isVisible = this._withinBounds(this._map.getPixelBounds(), this._bounds[this._map.getZoom()],this._map.options.crs.options.resolutions,this._map.getZoom());
+    this._bounds = this._getLayerBounds(this._groups,this._map.options.projection); //stores meter values of bounds
+    this.layerBounds = this._bounds[Object.keys(this._bounds)[0]];
+    for(let key of Object.keys(this._bounds)){
+      this.layerBounds.extend(this._bounds[key].min);
+      this.layerBounds.extend(this._bounds[key].max);
+    }
     L.GridLayer.prototype.onAdd.call(this,this._map);
+    this._map.fire('moveend',true);
   },
-
+  
   getEvents: function(){
     let events = L.GridLayer.prototype.getEvents.call(this,this._map);
-    events.moveend = this._setBoundsFlag;
+    this._parentOnMoveEnd = events.moveend;
+    events.moveend = this._handleMoveEnd;
+    events.move = ()=>{}; //needed to prevent moveend from running
     return events;
   },
 
 
   //sets the bounds flag of the layer and calls default moveEnd if within bounds
   //its the zoom level is between the nativeZoom and zoom then it uses the nativeZoom value to get the bound its checking
-  _setBoundsFlag : function(e){
-    let zoomLevel = this._map.getZoom();
+  _handleMoveEnd : function(e){
+    let mapZoom = this._map.getZoom();
+    let zoomLevel = mapZoom;
     zoomLevel = zoomLevel > this.options.maxNativeZoom? this.options.maxNativeZoom: zoomLevel;
     zoomLevel = zoomLevel < this.options.minNativeZoom? this.options.minNativeZoom: zoomLevel;
-    this.isVisible = this._withinBounds(this._map.getPixelBounds(), this._bounds[zoomLevel],this._map.options.crs.options.resolutions, this._map.getZoom());
-    if(!(this.isVisible))return;
-    this.fire('moveend',e,true);
```

Comment:
Why is this necessary?  In the code below you invoke the parent moveend handler directly.  Is this a relic, or intentional?